src/euphorie/client/browser/country.py

```diff
@@ -382,17 +382,50 @@ def form_action_name(self):
     def template(self):
         return self.index
 
+    @property
+    def organisation_options(self):
+        organisation_view = api.content.get_view(
+            "organisation", self.context, self.request
+        )
+        return [
+            {
+                "label": organisation.title,
+                "value": organisation.owner_id,
+                "selected": "selected"
+                if organisation.owner_id == int(self.request.get("organisation", ""))
+                else None,
+            }
+            for organisation in organisation_view.organisations
+        ]
+
+    def is_filter_active(self):
+        """True if any filters in the request parameters are different from the
+        defaults.
+        """
+        return (
+            self.request.get("organisation")
+            or self.request.get("sort_on", "alphabetical") != "alphabetical"
+        )
+
     @property
     @memoize
     def sessions(self):
         searchable_text = self.request.get("SearchableText", None)
         if searchable_text and "%" not in searchable_text:
             searchable_text = "%{}%".format(searchable_text)
+        sort_on_value = self.request.get("sort_on", "alphabetical")
+        if sort_on_value == "alphabetical":
+            order_by = self.webhelpers.survey_session_model.title
+        else:
+            order_by = False
+        organisation_value = self.request.get("organisation", "")
         return self.webhelpers.get_sessions_query(
             context=self.context,
             searchable_text=searchable_text,
             include_archived=True,
```

Comment:
Thinking out loud: having the include archived as a filter might be a nice enhancement. Anyway this is not in the scope of this task.
CC @pilz